backend/app/services/alpha_vantage_client.py

```
@@ -0,0 +1,96 @@
+            if not report_date_str:
+                continue
+            try:
+                report_date = datetime.strptime(report_date_str, "%Y-%m-%d").date()
+                if report_date >= today:
+                    future_dates.append(report_date)
+            except ValueError:
+                continue
+
+        result = min(future_dates).strftime("%Y-%m-%d") if future_dates else None
+        _cache[symbol] = (result, now)
+        return result
```

Ruff (0.15.4)

[warning] Consider moving this statement to an `else` block

(TRY300)